### ISSUE TITLE
stats calculator seed code changes between step 33 and 34 without explanation

### ISSUE THREAD

**Issue by hbar1st** [freeCodeCamp/freeCodeCamp#55689]
### Describe the Issue

The js beta project to build a statistics calculator has an unexpected change in the editor's code between step 33 and step 34. No explanation is given for this change.
In step 33 we are initially shown a forEach statement which looks like this:
```
  array.forEach(el => {

  })
```
We are asked to fill it in. Since back in step 29 we were shown sample code to help us write the new forEach, we probably end up writing it in a very similar way to the sample code there: https://www.freecodecamp.org/learn/javascript-algorithms-and-data-structures-v8/learn-advanced-array-methods-by-building-a-statistics-calculator/step-29

Then in step 34, the forEach suddenly becomes this one liner:
`array.forEach(el => counts[el] = (counts[el] || 0) + 1)`

But it is not acknowledged at all that the code changed. We don't even notice it till step 35 when it returns back to the main editing area.

I've seen people on the forum copy this code verbatim for other projects. However, I don't believe they know what it is doing (other than it counts the number of occurrences that is). Most people still struggle with truthies at this point of their learning so if we really want to write the code like this (and if we do, what was the point of step 29?), it would be great if there was a step between 33 and 34 that explains this shorthand.

### Affected Page

https://www.freecodecamp.org/learn/javascript-algorithms-and-data-structures-v8/learn-advanced-array-methods-by-building-a-statistics-calculator/step-34

### Your code

N/A

### Expected behavior

code campers write should not change so fundamentally without explanation in the project

### System

N/A

**Comment by lasjorg** [comment]
We deliberately changed it to the more verbose and easier to read/understand version at one point but I guess the seed code never changed for all the steps.

I will say, for any step where we allow all valid solutions to pass, it is expected that the seed code in the next step may not match what was written by the camper.

> code campers write should not change so fundamentally without explanation in the project

I don't think there is any way we can make sure of that and still allow all valid solutions.

**Comment by hbar1st** [comment]
Changing the seed code to match would be acceptable in my eyes. Teaching the shorthand code would also be great.
(why not just add a step or two to rewrite the long form which is definitely easier to understand to the short one? I think that would be worthwhile)

As additional context, I went to the forum and found people who noticed the change and didn't understand it:
https://forum.freecodecamp.org/t/learn-advanced-array-methods-by-building-a-statistics-calculator-step-34/697800
https://forum.freecodecamp.org/t/learn-advanced-array-methods-by-building-a-statistics-calculator-step-34/674808

I would encourage adding the steps to explain the code because it will re-enforce previous learning on truthies. (but again, if the seed code just changes to match, that's fine too)

**Comment by hbar1st** [comment]
an interesting thing I just found out: in the shopping cart project (much later than the calculator), step 24 actually attempts (fairly weakly) to teach a very similar statement https://www.freecodecamp.org/learn/javascript-algorithms-and-data-structures-v8/learn-basic-oop-by-building-a-shopping-cart/step-24
They do it like this:
>You now have a small bug. When you try to access a property of an object and the property doesn't exist, you get undefined. This means if the dessert isn't already present in the totalCountPerProduct object, you end up trying to add 1 to undefined, which results in NaN.

>To fix this, you can use the || operator to set the value to 0 if it doesn't exist. Wrap your right-hand totalCountPerProduct[dessert.id] in parentheses, and add || 0 to the end of the expression.

I thought I should mention it because the statement style is identical.

**Comment by Sembauke** [comment]
Hey @hbar1st,

It would be nice to teach the shorthand and add an extra step after step 33. To write the shorthand instead of the longer version.

**Comment by hbar1st** [comment]
I do not consider myself to be a capable JS teacher. I would be happy to help review any instructions from the perspective of the learner though.

**Comment by Supravisor** [comment]
Step 34 the `getMode` function sees the unexpected appearance of:

`console.log(counts)`

The `calculate` function already contains `console.log(getMode(numbers));`

In step 36 `console.log(counts)` disappears from `getMode`

Any idea why there are two `console.logs()` ?

**Comment by hbar1st** [comment]
i've had a recent experience with someone who didn't understand the new seed code and when I tried to explain, I gained some insight into areas that need to be clarified in the statement as it is being built out.
Specifically, the way the OR works as a kind of shortcut. It evaluates the statement on the left and if true returns the result but if not true, it just returns the expression on the right. 
`array.forEach(el => counts[el] = (counts[el] || 0) + 1)`

The explanation of the behaviour of the or is missing in the curriculum so far, so I would make sure to mention this part.
Another area to possibly mention again is truthys and falsys.  I think the camper didn't know that undefined is a falsy so thought the OR expression would evaluate to undefined somehow.

Just commenting in case these two points help flesh out the explanation better.
